catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/accesscontrol/AccessControlPreQueryPluginTest.java

```diff
@@ -0,0 +1,305 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.security.accesscontrol;
+
+import static org.codice.ddf.catalog.ui.forms.data.AttributeGroupType.ATTRIBUTE_GROUP_TAG;
+import static org.codice.ddf.catalog.ui.forms.data.QueryTemplateType.QUERY_TEMPLATE_TAG;
+import static org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants.WORKSPACE_TAG;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.Security;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.StopProcessingException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import org.geotools.filter.visitor.DuplicatingFilterVisitor;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.filter.Filter;
+
+public class AccessControlPreQueryPluginTest {
+  private static final Set<String> ACCESS_CONTROLLED_TAGS =
+      ImmutableSet.of(WORKSPACE_TAG, QUERY_TEMPLATE_TAG, ATTRIBUTE_GROUP_TAG);
+
+  private static final String ROLE_A = "A";
+
+  private static final String ROLE_B = "B";
+
+  private static final String EMAIL_BOB = "bob@example.net";
+
+  private static final String TEXT_CRITERIA = "*criteria*";
+
+  private static final String TEXT_ID = "123456578901";
+
+  private static final String TEXT_WILDCARD = "*";
+
+  private static final Date TEST_DATE = new Date();
+
+  private static final FilterBuilder FILTER_BUILDER = new GeotoolsFilterBuilder();
+
+  private static final DuplicatingFilterVisitor COPYING_VISITOR = new DuplicatingFilterVisitor();
+
+  private AccessControlPreQueryPluginUnderTest plugin;
+
+  @Before
+  public void setUp() {
+    plugin =
+        new AccessControlPreQueryPluginUnderTest(
+            new GeotoolsFilterBuilder(), // don't assert against things made by the same builder,
```

Comment:
We want to ensure we're dealing with the `Filter` definition of equality for these tests. Since builders are responsible for the impl that leaves them I didn't want to make any assumptions. If the builder impl changes on one side (prod or test) and things break, that is a symptom that something deeper is wrong. I didn't want the tests to start passing over time just because we were getting lucky when the builder impl changes on both sides. 

I'm not sure how to capture that concept in the code though.